File: SignalRApi/SignalR.Api.Model/ApiResponseModel.cs
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;

namespace SignalR.Api.Model;
public class ApiResponseModel
{
    [JsonPropertyName("isSuccess")]
    [NotNull]
    public bool IsSuccess { get; set; }

    [JsonPropertyName("statusCode")]
    [NotNull]
    public int StatusCode { get; set; }

    [JsonPropertyName("message")]
    [MaybeNull]
    public string Message { get; set; }

    [JsonPropertyName("data")]
    [MaybeNull]
    public object? Data { get; set; }

    [JsonPropertyName("error")]
    [MaybeNull]
    public ApiResponseErrorModel? Error { get; set; }

    [JsonPropertyName("totalPages")]
    [NotNull]
    public int TotalPages { get; set; }

    public ApiResponseModel(int statusCode, string messageKey, object data, int id = 0, int totalPage = 0)
    {
        var messageDto = LocalizedErrorCodes.GetErrorMessage(messageKey);
        this.StatusCode = statusCode;
        this.Message = messageDto.EnglishErrorMessage;
        this.Data = data;
        this.IsSuccess = true;
        this.TotalPages = totalPage;
    }

    public ApiResponseModel(int htppStatusCode, bool isSuccess, string messageKey, string message = "", string? stackTrace = null, int id = 0, int totalPage = 0)
    {
        var messageDto = LocalizedErrorCodes.GetErrorMessage(messageKey);
        this.IsSuccess = isSuccess;
        this.StatusCode = htppStatusCode;
        this.Error = new ApiResponseErrorModel(messageKey, messageDto.EnglishErrorMessage, messageDto.UdruErrorMessage);
        this.TotalPages = totalPage;
        this.Message = string.IsNullOrEmpty(message) ? messageDto.EnglishErrorMessage : message;
    }
}

File: signal_r_app/lib/repositories/models/api_response_model.dart
class ApiResponseModel<T> {
  late bool isSuccess;
  late int statusCode;
  late int totalPages;
  late String message;
  T? data;
  Error? error;

  ApiResponseModel(
      this.isSuccess, this.statusCode, this.totalPages, this.message,
      {this.data, this.error});

  ApiResponseModel.fromJson(Map<String, dynamic> json) {
    isSuccess = json['isSuccess'];
    statusCode = json['statusCode'];
    message = json['message'];
    data = json['data'];
    error = json['error'] != null ? Error.fromJson(json['error']) : null;
    totalPages = json['totalPages'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['isSuccess'] = isSuccess;
    data['statusCode'] = statusCode;
    data['message'] = message;
    data['data'] = this.data;
    if (error != null) {
      data['error'] = error!.toJson();
    }
    data['totalPages'] = totalPages;
    return data;
  }
}

class Error {
  late String errorCode;
  late String englishMessage;
  String? urduMessage;
  String? stackTrace;

  Error(this.errorCode, this.englishMessage,
      {this.urduMessage, this.stackTrace});

  Error.fromJson(Map<String, dynamic> json) {
    errorCode = json['errorCode'];
    englishMessage = json['englishMessage'];
    urduMessage = json['urduMessage'];
    stackTrace = json['stackTrace'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['errorCode'] = errorCode;
    data['englishMessage'] = englishMessage;
    data['urduMessage'] = urduMessage;
    data['stackTrace'] = stackTrace;
    return data;
  }
}
